File: Chapter 05/JS/section_5_6_crypto_tracker/before/attemptedTest.js
const PortfolioManager = require('./portfolioManager');

/**
 * A demonstration of why Tight Coupling ruins testability.
 */
class AttemptedTest {
    static async run() {
        console.log("\n--- ATTEMPTING TO TEST (BEFORE) ---");
        
        const manager = new PortfolioManager();

        console.log("Test Action: Calculating value of 1 BTC...");
        
        try {
            const value = await manager.calculateTotalValue(1.0);
            
            // ASSERT
            // We cannot write a reliable test assertion because the data is live.
            console.log(`Result: ${value}`);
            console.log("FAIL: This test is FLAKY. We cannot assert a fixed price.");
        } catch (error) {
            console.log("CRASH: Test failed completely. No internet connection or API is down.");
        }
    }
}

module.exports = AttemptedTest;

File: Chapter 05/Python/section_5_6_crypto_tracker/before/attempted_test.py
from .portfolio_manager import PortfolioManager
from shared.log_manager import LogManager

class AttemptedTest:
    """A demonstration of why Tight Coupling ruins testability."""
    
    @staticmethod
    def run() -> None:
        LogManager.info("AttemptedTest", "\n--- ATTEMPTING TO TEST (BEFORE) ---")
        
        manager = PortfolioManager()

        LogManager.info("AttemptedTest", "Test Action: Calculating value of 1 BTC...")
        
        try:
            value = manager.calculate_total_value(1.0)
            
            # ASSERT
            # We cannot assert equality because the live price is unpredictable.
            LogManager.info("AttemptedTest", "Result: {0}", value)
            LogManager.info("AttemptedTest", "FAIL: This test is FLAKY. We cannot assert a fixed price.")
        except Exception:
            LogManager.info("AttemptedTest", "CRASH: Test failed completely. No internet connection or API down.")